src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -54,14 +54,14 @@ public class JsonRpc : IDisposableObservable
         /// A map of outbound calls awaiting responses.
         /// Lock the <see cref="dispatcherMapLock"/> object for all access to this member.
         /// </summary>
-        private readonly Dictionary<RequestId, OutstandingCallData> resultDispatcherMap = new Dictionary<RequestId, OutstandingCallData>();
+        private readonly Dictionary<long, OutstandingCallData> resultDispatcherMap = new Dictionary<long, OutstandingCallData>();
 
         /// <summary>
```

Comment:
Being internal means this magic trick isn't reproducible by other (external) formatters. I like that it's your default, but I'd like to explore whether we can make the functionality available publicly. 
Your method as it stands is very focused on your feature. If however we change it to simply return the `JsonRpcMethodAttribute` for *any* RPC method, then it will likely be useful not just for this but other future features, which makes me feel better about making it public.
I'm going to play around with this idea and may push a commit to your PR to see what you think, if that's OK with you.